paper.md

```diff
@@ -24,55 +24,74 @@ bibliography: paper.bib
+We also experimented with using biaxial biLSTM along the time and frequency axes as the bottleneck layers, but it took slightly longer to train yet offered a negligible improvement.
 
-## Danna-Sep
-Finally, we calculated the weighted average of the outputs obtained from the models mentioned above as our final output. Experiments are conducted to search for optimal weighting. The optimal weights for each source, input domain type (T for waveforms, TF for frequency masking), and the size of the models are given in the following table.
+## Demucs
+For Demucs [@defossez2019music], we built upon the variant with 48 hidden channels, and enhanced the model by replacing the decoder with four independent decoders responsible for four respective sources.
+Each decoder has the same architecture as the original decoder, except for size of the hidden channel which was reduced to 24. 
+This makes the total number of parameters comparable with the original Demucs.
+The training loss aggregates the L1-norm between estimated and ground-truth waveforms of the four sources.
+The model took approximately 10 days to train on a single RTX 3070 using mixed precision with a batch size of 16, and four steps of gradient accumulation.
 
-|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) |
-|---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:|
-| X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
+## Danna-Sep
+In order to obtain the final output of our framework, we calculated weighted average of individual outputs from the above-mentioned models.
+Experiments were conducted to search for optimal weighting. The optimal weights for each source, types of input domain (T for waveforms, TF for frequency masking), and the sizes of the models are given in the following table.
+|         | Drums | Bass | Other | Vocals | Input Domain | Size (Mb) | |---------|:-----:|:----:|:-----:|:------:|:------------:|:---------:| | X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
 | U-Net   | 0.2   | 0.17 | 0.5   | 0.4    | TF | 61
 | Demucs  | 0.6   | 0.73 | 0.5   | 0.4    | T | 733
 
 All models were trained on the training set of musdb18-hq [@musdb18-hq] using an Adam optmizier[@kingma2014adam]. 
 
 # Separation performances
-We evaluated our models in terms of Signal-to-Distortion Ratio (SDR) [@vincent2006performance] on musdb18 [@musdb18] using the *museval* toolbox [@fabian_robert_stoter_2019_3376621]. One iteration of MWF was used for X-UMX and U-Net, and we didn't apply the shift trick [@defossez2019music] for our Demucs model. The results of our models and the original X-UMX and Demucs are shown in the table below.
+We performed the benchmark using musdb18-hq [@musdb18] as the dataset for a fair comparison, and evaluated the models in terms of Signal-to-Distortion Ratio (SDR) [@vincent2006performance] derived from *museval* [@fabian_robert_stoter_2019_3376621].
```

Comment:
We trained all models on musdb18-hq (high quality) but the benchmark was done on musdb18 (low quality) so the score can be compared to other works. This is different from X-UMX and Demucs, so I add one more line to state that the baselines were also trained on high quality dataset.
I prefer the previous version.

Reply:
Thanks for clarifying; refer to [commit](https://github.com/yoyololicon/mdx-submissions21/pull/4/commits/09ada99f83b65e567bf7c99a9d85371155d76190) for the update.